src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -135,4 +140,16 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
           return
         #set isInit to true to force redraw after all updates are processed
         child.updateModel model
+
+
+      modelsLength: =>
+        return 0 unless @scope.models?
+
+        if @scope.models != @cached.models
+          @cached.models = @scope.models
+          array = if angular.isArray(@scope.models) then @scope.models else Object.keys(@scope.models)
+          @cached.modelsLength = array.length
+        #we have existing cachedLength or one was just set
+        return @cached.modelsLength
```

Comment:
Will this break if someone modifies `@scope.models` (rather than assigns a new object or array)?  It looks to me like pushing new models onto an array (or adding new models to an object) wouldn't trigger update of `@cached.modelsLength`.